fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -1446,4 +1464,21 @@ public Pair<Loan, Map<String, Object>> assembleLoanApproval(AppUser currentUser,
 
         return Pair.of(loan, actualChanges);
     }
+
+    private boolean validateDownPaymentAttribute(final boolean isDownPaymentEnabledInLoanProduct, final JsonElement element) {
```

Comment:
Please move the validations into the Validator class!

Reply:
Done! moved to the validator